examples/querying_entity_traits.ipynb

```diff
@@ -0,0 +1,710 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0fcf0214-2561-488a-8eee-0330ca27710d",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "entity_trait_set = manager.entityTraits(logo_ref, EntityTraitAccess.kRead, context)\n",
+    "\n",
+    "helpers.display_result(entity_trait_set)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a6c4f0ff-0e5d-40b7-84d6-d46efa8cba7a",
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": []
+   },
+   "source": [
+    "`EntityTraitAccess.kRead` tells the manager the reason why you want the trait set. In this case we want to use the result in a read context, e.g. `resolve`ing properties, grouping entities of the same type in a UI, or determining the type of entity after a drag-and-drop of an entity reference into the application.\n",
+    "\n",
+    "What happens when we `resolve` this trait set?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "entity_data = manager.resolve(logo_ref, entity_trait_set, ResolveAccess.kRead, context)\n",
+    "\n",
+    "helpers.display_result(entity_data)"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "fbae28c950c33589"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Note that not all traits are associated with `resolve`able properties - many are simply tags. However, it is safe to `resolve` the full trait set - those that do not have properties are simply missing from the response.\n",
+    "\n",
+    "As an aside, we can filter the trait set through `managementPolicy`"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "609f599ef0708376"
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "policy_data = manager.managementPolicy(entity_trait_set, PolicyAccess.kRead, context)\n",
+    "\n",
+    "helpers.display_result(policy_data)"
+   ],
+   "metadata": {
+    "collapsed": false
+   },
+   "id": "b5c27257bb1af339"
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Note how the trait set returned from `managementPolicy` (minus any `policy` traits) matches the trait set we successfully `resolve`d.\n",
+    "\n",
+    "In this way `managementPolicy` can be used to filter the trait set of an entity to only those traits that have `resolve`able properties.\n",
+    "\n",
+    "In practice, a well-behaved host will cherry-pick only those traits the host needs to `resolve`. That trait set should then be passed to `managementPolicy` to determine what the manager can actually provide. This process is independent of `entityTraits`. That is, the host should know at application startup which traits it's going to need, and so query `managementPolicy` ahead of time and store the result for use later. See relevant documentation for `managementPolicy` for more information."
```

Comment:
Might be good to highlight how `managementPolicy` is constant for any given manager, whereas `entityTraits` is per-entity. The first sentence might be interpreted if skimmed as meaning "do this each time before you resolve". Maybe highlight that it is optional?

Reply:
I do wonder if it'll turn out that manager implementers want policies to vary by entity reference. Something to watch out for in user feedback.

Reply:
Added a sentence in e82caf93